pygmt/base_plotting.py

```diff
@@ -318,24 +318,59 @@ def grdcontour(self, grid, **kwargs):
 
     @fmt_docstring
     @use_alias(
-        R="region",
-        J="projection",
-        W="pen",
+        A="img_out",
         B="frame",
-        I="shading",
         C="cmap",
+        D="img_in",
+        E="dpi",
+        G="bit_color",
+        I="shading",
+        J="projection",
+        M="monochrome",
+        N="no_clip",
+        Q="nan_transparent",
+        R="region",
+        U="timestamp",
+        V="verbose",
         X="xshift",
         Y="yshift",
+        n="interpolation",
         p="perspective",
         t="transparency",
         x="cores",
     )
     @kwargs_to_strings(R="sequence", p="sequence")
     def grdimage(self, grid, **kwargs):
         """
-        Project grids or images and plot them on maps.
-
-        Takes a grid file name or an xarray.DataArray object as input.
+        Project and plot grids or images.
+
+        Reads a 2-D grid file and produces a gray-shaded (or colored) map by
+        building a rectangular image and assigning pixels a gray-shade (or
+        color) based on the z-value and the CPT file. Optionally, illumination
+        may be added by providing a file with intensities in the (-1,+1) range
+        or instructions to derive intensities from the input data grid. Values
+        outside this range will be clipped. Such intensity files can be created
+        from the grid using `grdgradient` and, optionally, modified by
+        `grdmath` or `grdhisteq`. A third alternative is available when GMT is
+        build with GDAL support. Pass **image** which can be an image file
+        (geo-referenced or not). In this case the image can optionally be
```

Comment:
Agree that this could be worded better, currently it's just copied from upstream GMT at https://github.com/GenericMappingTools/gmt/blob/8c0b548e52938f91bfb15f47f27d023488003ef8/doc/rst/source/grdimage_common.rst_#L14-L16. Maybe we should take out the `grdgradient`, `grdmath` and `grdhisteq` parts since they're not wrapped in PyGMT yet?

Reply:
I'm afraid we may forget to add them back when the wrappers are available.

Reply:
Maybe keep them for now, better to over-document than not. Doubt we'll see `grdmath` anytime soon (who uses Reverse Polish notation?!!), but `grdgradient` would definitely come, and `grdhisteq` will be a nice one for stretching colormaps.

Reply:
Yes, I agree.